app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -72,7 +80,33 @@ protected void onCreate(Bundle savedInstanceState) {
         downloadProgress.setVisibility(View.VISIBLE);
         downloadProgress.setIndeterminate(true);
         DataDownload download = new DataDownload();
-        download.downloadVersions();
+        ConnectivityManager connectivityManager = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
+        if(networkInfo!=null && networkInfo.isConnected()){
+            download.downloadVersions();
+        }
+        else {
+            downloadProgress.setVisibility(View.GONE);
+            AlertDialog.Builder builder = new AlertDialog.Builder(this);
+            builder.setTitle("Network not available")
+                    .setMessage("Turn on WiFi or Mobile data in Settings")
+                    .setPositiveButton("Settings", new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialog, int which) {
+                            Intent setNetworkIntent = new Intent(Settings.ACTION_SETTINGS);
+                            startActivity(setNetworkIntent);
+                            MainActivity.this.finish();
+                        }
+                    })
+                    .setNeutralButton("Cancel", new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialog, int which) {
+
```

Comment:
Why?

The purpose of MainActivity.this.finish() is to close the app as soon as I go to Settings, otherwise download won't happen even after switching WiFi/Data on.